pkg/hive/manager.go

```diff
@@ -53,61 +60,49 @@ func NewFromConfig(log *logrus.Entry, subscriptionDoc *api.SubscriptionDocument,
 		return nil, err
 	}
 
-	return new(subscriptionDoc, doc, hiveclientset, kubernetescli, dh), nil
+	return newClusterManager(log, _env, clusterUUID, subscriptionID, hiveclientset, kubernetescli, dh), nil
 }
 
-func new(subscriptionDoc *api.SubscriptionDocument, doc *api.OpenShiftClusterDocument, hiveClientset *hiveclient.Clientset, kubernetescli *kubernetes.Clientset, dh dynamichelper.Interface) ClusterManager {
+// newClusterManager creates a ClusterManager. It takes components from CosmosDB
+// documents that do not change such as the cluster's UUID and the subscription
+// UUID. Values that can change (such as the subscription tenant and values in
+// OpenShiftCluster itself) are given in the CreateOrUpdate/Install methods
+// since passing the OpenShiftClusterDocument here does not mean that it is
+// updated when the document itself is later amended.
+func newClusterManager(log *logrus.Entry, _env env.Core, clusterUUID string, subscriptionID string, hiveClientset *hiveclient.Clientset, kubernetescli *kubernetes.Clientset, dh dynamichelper.Interface) ClusterManager {
```

Comment:
It probably doesn't make much sense to have `clusterUUID` and `subscriptionID` in here then. Let's pass cluster doc and sub doc into `CreateOrUpdate` and where else we need it.